libraries/engage/product/hocs/withProductListEntryProduct.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import withProductListEntry from './withProductListEntry';
+import { getProduct } from '../selectors/product';
+
+/**
+ * Injects the product from the ProductListEntryContext into the wrapped component. The product
+ * property name can be configured via the HOC options.
+ * @param {Function} WrappedComponent The react component to wrap.
+ * @param {Object} [options={}] Options for the HOC.
+ * @param {string} [options.prop="product"] An optional prop name to inject the
+ * product into the wrapped component.
+ * @returns {JSX}
```

Comment:
my IDE suggests this function doesn't actually return JSX but rather returns {function(*): *}

Reply:
changed